packages/wallet-lib/src/types/Storage/methods/rehydrateState.js

```diff
@@ -16,27 +14,45 @@ const rehydrateState = async function rehydrateState() {
       if (this.adapter && hasMethod(this.adapter, 'getItem')) {
         const wallets = await this.adapter.getItem('wallets');
         if (wallets) {
-          wallets.forEach((walletState) => {
-            const walletStore = new WalletStore();
-            walletStore.importState(walletState);
-            this.wallets.set(walletStore.walletId, walletStore);
-          });
+          try {
+            Object.keys(wallets).forEach((walletId) => {
+              const walletStore = this.getWalletStore(walletId);
+
+              if (walletStore) {
+                walletStore.importState(wallets[walletId]);
+              }
+            });
+          } catch (e) {
+            logger.error('Error importing wallets storage, resyncing from start', e);
+
+            this.adapter.setItem('wallets', null);
```

Comment:
Shouldn't we propagate this error to wallet error event?

Reply:
Not really. For now we don't expect users to react in any way on this error.

Reply:
I see. Can we catch specific errors here, otherwise event syntax and other errors will be treated as an import error?

Reply:
Syntax error won't pass because tests will fail, as with all other errors - we expect this function to catch as many things as possible for now.